keps/sig-storage/5040-remove-gitrepo-driver/README.md

```diff
@@ -0,0 +1,966 @@
+should help to ensure that the tone and content of the `Summary` section is
+useful for a wide audience.
+
+A good summary is probably at least a paragraph in length.
+
+Both in this section and below, follow the guidelines of the [documentation
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+
+We propose removing support for the in-tree gitRepo volume driver.
+
+We acknowledge that this is highly unusual, but there are mitigating circumstances:
+* gitRepo volume types can be exploited to gain remote code execution as root on the nodes as shown in [CVE-2024-10220](https://nvd.nist.gov/vuln/detail/cve-2024-10220)
+* gitRepo has perfectly workable alternatives:
+  * Using [git-sync](https://github.com/kubernetes/git-sync)
+  * Using an initContainer. See https://kubernetes.io/docs/concepts/storage/volumes/#gitrepo for more details
+* gitRepo has been deprecated for a long time and is unmaintained
+* gitRepo has low usage (based on limited data)
+* there exists precedent for removing volume drivers which were unmaintained like [glusterfs](https://kubernetes.io/docs/concepts/storage/volumes/#glusterfs)
+
+Given this, we think kubernetes should EOL this feature with urgency.
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+Remove a vector for remote code execution through an ancient and unmaintained
+volume plugin.
+
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+- Remove in-tree gitRepo volume support.
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+## Proposal
+
+Kubernetes drops support for gitRepo volume types by removing the in-tree driver
+that actually mounts the volume in the pod. In this case we will still keep the
+gitRepo volume in the API. This removal is similar to removal of support for
+[glusterfs](https://kubernetes.io/docs/concepts/storage/volumes/#glusterfs)
+which was deprecated in v1.25 and removed in v1.26.
+
+Sig-storage was actually very close to doing this and had even added a notice of
+removal in the release notes for v1.32. But due to concerns raised by sig-arch
+they rolled back the notice. sig-storage provided [this rationale](https://github.com/kubernetes/kubernetes/issues/125983#issuecomment-2522201283) for considering removal.
```

Comment:
(nit)
```suggestion
SIG Storage was actually very close to doing this and had even added a notice of
removal in the release notes for v1.32. But due to concerns raised by SIG Architecture
they rolled back the notice. sig-storage provided [this rationale](https://github.com/kubernetes/kubernetes/issues/125983#issuecomment-2522201283) for considering removal.
```

This is however motivation, not the proposed change.

Reply:
Fixed the nit and moved to motivation.